### PR TITLE
MG-888 - Update channels SDK tests

### PR DESCRIPTION
**Pull request by felixgateru**
# What type of PR is this?
This is a feature as it updates the channels SDK tests.

## What does this do?
This pr updates teh current tests in the sdk to conform to existing changes as well as increasing the coverage of the channels sdk tests.

## Which issue(s) does this PR fix/relate to?

- Related Issue #888 

## Have you included tests for your changes?
Yes, the tests are the changes.

## Did you document any new/modified feature?
No,

### CODE REVIEW

**Review by arvindh123 — DISMISSED** [review]

File: pkg/sdk/go/channels.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"net/http"
 	"time"
 
+	"github.com/absmach/magistrala/internal/apiutil"
```
Comment (rodneyosodo) [inline pkg/sdk/go/channels.go]: Do the ID check for `ChannelsByThing`, `ChannelPermissions`, `ChannelPermissions`, `ChannelPermissions`,`ListChannelUsers`, `ListChannelUsers`, `ListChannelUsers`, `ListChannelUsers`, `ConnectThing`, `ConnectThing`, (`ConnectThing`, `ConnectThing` -> inside changeChannelStatus)

File: pkg/sdk/go/channels_test.go [hunk 2]
```diff
@@ -4,159 +4,238 @@
 package sdk_test
 
 import (
-	"context"
 	"fmt"
 	"net/http"
 	"net/http/httptest"
 	"reflect"
+	"strings"
```
Comment (rodneyosodo) [inline pkg/sdk/go/channels_test.go:10]: Add test for `ListChannelUsers`

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: pkg/sdk/go/channels.go (hunk 1 above)
Comment (felixgateru) [reply pkg/sdk/go/channels.go]: These checks already exist on the api level

File: pkg/sdk/go/channels_test.go (hunk 2 above)
Comment (felixgateru) [reply pkg/sdk/go/channels_test.go:10]: Added to users tests. Will be refactored and relocated by #2195

**Review by rodneyosodo — DISMISSED** [review]

**Review by arvindh123 — DISMISSED** [review]

**Review by rodneyosodo — DISMISSED** [review]

**Review by dborovcanin — APPROVED** [review]

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**